### DISCUSSION TITLE
"Overlay against underlay" discussion in Dual stacks section

### DISCUSSION THREAD

**Discussion by EduardVasilenko** [becarpenter/book6#40] [General]
Hi Brian,
Independent where this discussion would be moved,
it should have a reference to RFC 8950 - it is a critical piece of this discussion.
Because just 2 years ago it was not possible to organize IPv4 VPN (overlay) over IPv6 underlay.
Eduard

**Comment by becarpenter** [comment]
Should draft-xie-v6ops-framework-md-ipv6only-underlay also discuss RFC 8950? I think there is a lot of material in that draft that is interesting for us. I won't make any changes right now, but this needs work to present the complete picture.

**Comment by EduardVasilenko** [comment]
I do not understand draft-xie-v6ops-framework-md-ipv6only-underlay at all. Hence, I am silent on it.
It looks like Chongfeng would like to convert DualStack Overlay with the Internet (and maybe other Overlays/VPNs in the future) to IPv6-only in some way.
The offline discussion did not help me to understand how. Initally, he was talking about stateful translation (464XLAT).
The latest that I got from Chongfeng is to swamp all IPv4 prefixes (900k) into IPv6 table (150k now) by prepending a prefix (stateless translation).
It wouldn't work technically (prefixes would occupy 4x memory). And it does not have a business justification.
Ed/

**Comment by becarpenter** [comment]
In fact a primary design goal for IPv6 routing to was to keep the IPv4 swamp out of the IPv6 BGP-4 table!

**Comment by becarpenter** [comment]
Finally got to this: added a pointer to RFC8950 in [Routing](https://github.com/becarpenter/book6/blob/main/2.%20IPv6%20Basic%20Technology/Routing.md) section.
